code/game/objects/items/implants/implant_abductor.dm

```diff
@@ -13,22 +13,51 @@
 		to_chat(imp_in, span_warning("You must wait [timeleft(on_cooldown)*0.1] seconds to use [src] again!"))
 		return
 
-	home.Retrieve(imp_in,1)
+	if(isnull(home) && !link_pad())
+		if(!link_pad())
+			imp_in.balloon_alert(imp_in, "no teleport pads detected!")
+			return
```

Comment:
Why run link_pad() twice? Cache it to a variable or something so we don't have to keep rerunning the proc. You also probably want to have an early return for isnull(home) too? I see where you're going but running link_pad twice is putting my head in a spin